core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/FiltersAggregatorBuilder.java

```diff
@@ -57,6 +58,8 @@ public FiltersAggregatorBuilder(String name, KeyedFilter... filters) {
 
     private FiltersAggregatorBuilder(String name, List<KeyedFilter> filters) {
         super(name, InternalFilters.TYPE);
+        // internally we want to have a fixed order of filters, regardless of the order of the filters in the request
+        Collections.sort(filters);
```

Comment:
since there is no natural order for filters, I would rather not make KeyedFilter implement Comparable and provide a comparator to Collections.sort?

also I'd prefer if we copied the list instead of modifying it in place